roofit/roofitcore/src/RooFactoryWSTool.cxx

```diff
@@ -716,7 +716,7 @@ RooAddition* RooFactoryWSTool::addfunc(const char *objName, const char* specList
 
   sum->setStringAttribute("factory_tag",Form("sum::%s(%s)",objName,specList)) ;
   if (_ws->import(*sum,Silence())) logError() ;
-  return (RooAddition*) _ws->pdf(objName) ;
+  return static_cast<RooAddition*>(_ws->function(objName));
```

Comment:
Here in this case, `clang-tidy` was even catching an invalid cast. Before, `RooAbsPdf*` was casted to `RooAddition*`, which is not allowed because `RooAddition` is not a pdf.